src/components/selectable/selectable_list/selectable_list.tsx

```diff
@@ -394,6 +454,123 @@ export class EuiSelectableList<T> extends Component<EuiSelectableListProps<T>> {
     );
   };
 
+  forceVirtualizedListRowRerender = () => {
+    this.setState({ itemData: { ...this.state.optionArray } });
+  };
+
+  // EuiTextTruncate is expensive perf-wise - we use several utilities here to
+  // offset its performance cost
+
+  // and creates a resize observer for
+  // each individual item. This logic tries to offset this performance hit by
+  // guesstimating a default width for each option
+  focusBadgeOffset = 0;
+
+  calculateDefaultOptionWidth = ({
+    width: containerWidth,
+  }: EuiAutoSizeHorizontal) => {
+    const { truncationProps, searchable, searchValue } = this.props;
+
+    // If it's not likely we'll need to use EuiTextTruncate, don't set state/rerender on every panel resize
+    const mayTruncate = searchable || truncationProps;
+    if (!mayTruncate) return;
+
+    const paddingOffset = this.props.paddingSize === 'none' ? 0 : 24; // Defaults to 's'
+    const checkedIconOffset = this.props.showIcons === false ? 0 : 28; // Defaults to true
+    this.focusBadgeOffset = this.props.onFocusBadge === false ? 0 : 46;
+
+    this.setState({
+      defaultOptionWidth: containerWidth - paddingOffset - checkedIconOffset,
+    });
+
+    // Potentially force list rows to rerender on dynamic resize as well,
+    // but try to do it as lightly as possible
+    if (truncationProps) {
+      this.forceVirtualizedListRowRerender();
+    } else if (searchable && searchValue) {
+      this.forceVirtualizedListRowRerender();
+    }
```

Comment:
I think this would be more DRY. What do you say?
```suggestion
    if (truncationProps || (searchable && searchValue)) {
      this.forceVirtualizedListRowRerender();
    }
```

Reply:
for sure!